src/lib/storage/lz4/lz4_iterable.hpp

```diff
@@ -22,24 +22,48 @@ class LZ4Iterable : public PointAccessibleSegmentIterable<LZ4Iterable<T>> {
 
     auto decompressed_segment = _segment.decompress();
 
-    auto begin = Iterator<ValueIterator>{decompressed_segment.cbegin(), _segment.null_values().cbegin()};
-    auto end = Iterator<ValueIterator>{decompressed_segment.cend(), _segment.null_values().cend()};
+    /**
+     * If the null value vector doesn't exist, then the segment does not have any row value that is null. In that case,
+     * we can just use a default initialized boolean vector.
+     */
+    const auto null_values =
+        _segment.null_values().has_value() ? _segment.null_values().value() : pmr_vector<bool>(_segment.size());
```

Comment:
Why did you change this? `value()` makes no sense if you have already checked for its existence. How about
`const auto null_values = _segment.null_values() ? *_segment.null_values() : pmr_vector<bool>(_segment.size());

Reply:
To keep it consistent with the other cases of accessing an `std::optional`'s value.

Reply:
In the rest of the code base, there are more cases of the implicit bool and the dereference operator being used. Even better, I just remembered that `optional` has `value_or`:

`const auto null_values = _segment.null_values().value_or(pmr_vector<bool>(_segment.size()))`

(untested)

Reply:
Ah, wait. That one would probably always create the vector. Maybe

`const auto null_values = _segment.null_values().value_or(_segment.size())`?

Haven't worked with that myself yet.

Reply:
But then, grepping through the code base, I find more of those unnecessary uses of `value()`. Apparently, it's not as consistent as I hoped it would be.